src/main/java/com/couchbase/lite/support/Batcher.java

```diff
@@ -55,7 +65,7 @@ public Batcher(ScheduledExecutorService workExecutor,
         this.processor = processor;
         this.inbox = new ArrayList<T>();
         this.scheduled = false;
-        this.lastProcessedTime = System.currentTimeMillis();
+        this.lastProcessedTime = 0;
```

Comment:
As discussed, setting lastProcessedTime to the current time here will result to unpredictable result depending on time to create the batcher and time to queue the object. For more info, iOS also sets the value to zero and let the schedule objects to decide what the delay should be.